File: Lexor/UI/lexor_mobile/lib/models/salary_slip_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'salary_slip_response.g.dart';

@JsonSerializable(createToJson: false)
class SalarySlipUserResponse {
  final String firstName;
  final String lastName;

  SalarySlipUserResponse({required this.firstName, required this.lastName});

  factory SalarySlipUserResponse.fromJson(Map<String, dynamic> json) =>
      _$SalarySlipUserResponseFromJson(json);
}

@JsonSerializable(createToJson: false)
class SalarySlipEmployeeResponse {
  final SalarySlipUserResponse user;

  SalarySlipEmployeeResponse({required this.user});

  factory SalarySlipEmployeeResponse.fromJson(Map<String, dynamic> json) =>
      _$SalarySlipEmployeeResponseFromJson(json);
}

@JsonSerializable(createToJson: false)
class SalarySlipItemResponse {
  final int itemType;
  final String name;
  final String? description;
  final double? quantity;
  final double? rate;
  final double? multiplier;
  final double amount;

  SalarySlipItemResponse({
    required this.itemType,
    required this.name,
    this.description,
    this.quantity,
    this.rate,
    this.multiplier,
    required this.amount,
  });

  factory SalarySlipItemResponse.fromJson(Map<String, dynamic> json) =>
      _$SalarySlipItemResponseFromJson(json);
}

@JsonSerializable(createToJson: false)
class SalarySlipResponse {
  final int id;
  final int year;
  final int month;
  final double brutoSalary;
  final double adjustedBruto;
  final double totalContributions;
  final double taxBase;
  final double tax;
  final double netSalary;
  final int status;
  final DateTime generatedAt;
  final DateTime? paidAt;
  final SalarySlipEmployeeResponse employee;
  final List<SalarySlipItemResponse>? items;

  SalarySlipResponse({
    required this.id,
    required this.year,
    required this.month,
    required this.brutoSalary,
    required this.adjustedBruto,
    required this.totalContributions,
    required this.taxBase,
    required this.tax,
    required this.netSalary,
    required this.status,
    required this.generatedAt,
    this.paidAt,
    required this.employee,
    this.items,
  });
  String get employeeFullName =>
      '${employee.user.firstName} ${employee.user.lastName}';

  factory SalarySlipResponse.fromJson(Map<String, dynamic> json) =>
      _$SalarySlipResponseFromJson(json);
}

File: Lexor/Lexor.Model/Responses/SalarySlipResponse.cs
using Lexor.Model.Enums;

namespace Lexor.Model.Responses
{
    public class SalarySlipResponse
    {
        public int Id { get; set; }
        public EmployeeResponse Employee { get; set; } = null!;
        public int Year { get; set; }
        public int Month { get; set; }
        public decimal BrutoSalary { get; set; }
        public decimal AdjustedBruto { get; set; }
        public decimal TotalContributions { get; set; }
        public decimal TaxBase { get; set; }
        public decimal Tax { get; set; }
        public decimal NetSalary { get; set; }
        public SalarySlipStatus Status { get; set; } = SalarySlipStatus.Pending;
        public DateTime GeneratedAt { get; set; }
        public DateTime? PaidAt { get; set; }
        public List<ItemResponse>? Items { get; set; } 

        public class ItemResponse
        {
            public int Id { get; set; }

            public SalarySlipItemType ItemType { get; set; }

            public string Name { get; set; } = string.Empty;

            public string? Description { get; set; }

            public decimal? Quantity { get; set; }

            public decimal? Rate { get; set; }

            public decimal? Multiplier { get; set; }

            public decimal Amount { get; set; }
        }

        public class EmployeeResponse
        {
            public int Id { get; set; }
            public UserResponse User { get; set; } = null!;
            // Filled only where Department is loaded (monthly report); null in list views.
            public string? DepartmentName { get; set; }
        }

        public class UserResponse
        {
            public int Id { get; set; }
            public string FirstName { get; set; } = string.Empty;
            public string LastName { get; set; } = string.Empty;

            // Small avatar so list views can show the person's photo next to their name
            // (guideline 6) without carrying the full-size image (guideline 8.2).
            public string? ProfileThumbnailBase64 { get; set; }
        }
    }
}
